data/config-schema/src/main/resources/net/corda/schema/configuration/crypto/1.0/corda.crypto.json

```diff
@@ -91,107 +92,46 @@
           },
           "additionalProperties": false
         },
-        "categories": {
-          "description": "Categories which the HSM can be used for",
+        "wrappingKeys": {
+          "description" : "Key derivation parameters for wrapping keys supplied in config",
           "type": "array",
-          "minItems": 1,
           "items": {
             "type": "object",
             "properties": {
-              "category": {
-                "description": "Category name, could be the wildcard * meaning all which has to be the last item if used",
+              "alias": {
+                "description": "The alias for the wrapping key.",
                 "type": "string"
               },
-              "policy": {
-                "description": "Defines how private keys are generated for that category, ALIASED - stored in the HSM, WRAPPED - wrapped by the HSM, BOTH - decision is taken by the HSM at runtime",
-                "enum": [
-                  "ALIASED",
-                  "WRAPPED",
-                  "BOTH"
-                ]
+              "algorithm": {
+                "description": "Key derivation function and wrapping key algorithm selection",
+                "type": "string",
+                "default": "PBKDF2WithHmacSHA256"
+              },
+              "salt": {
+                "description": "Salt for the key derivation function",
+                "type": "string"
+              },
+              "passphrase": {
+                "description": "Passphrase for the key derivation function",
+                "type": "string"
               }
             },
+            "additionalProperties": false,
             "required": [
-              "category",
-              "policy"
-            ],
-            "additionalProperties": false
+              "alias",
+              "salt",
+              "passphrase"
+            ]
           }
         },
-        "masterKeyPolicy": {
-          "description": "Wrapping key policy, NONE - no wrapping key is required, SHARED - wrapping key shared by all tenants, UNIQUE - wrapping key unique for each tenant",
-          "enum": [
-            "NONE",
-            "SHARED",
-            "UNIQUE"
-          ]
-        },
-        "masterKeyAlias": {
-          "description": "If the masterKeyPolicy is SHARED then this specifies the alias of the shared key and as such is required",
+        "defaultWrappingKey": {
+          "description": "The default wrapping key, which must be in the wrappingKeys array.",
```

Comment:
This is marked as type string which is not nullable. I think the integration test will fail if you dont define a default value such as the one below. As the integration test passes an empty config to the schema and asks for it to generate defaults. but as the field is "required" then it will fail validation as theres no default value or real value set.

"default": "",

Reply:
after some offline discussion with dickon and kyriakos, i think the path forward to to not set defaults for the required fields and rely on the config passed to the defaulter to have these fields set already